e2e/tests/rate_limiting/rate_limiting_test.go

```diff
@@ -0,0 +1,249 @@
+
+		// Balanced moved form useA to userB
+		s.Require().Equal(userABalBefore-testvalues.IBCTransferAmount, userABalAfter)
+		escrowBalA, err := query.Balance(ctx, chainA, escrowAddrA.String(), denomA)
+		s.Require().NoError(err)
+		s.Require().Equal(testvalues.IBCTransferAmount, escrowBalA.Int64())
+
+		userBBalAfter, err := query.Balance(ctx, chainB, userB.FormattedAddress(), ibcTokenB.IBCDenom())
+		s.Require().NoError(err)
+		s.Require().Equal(testvalues.IBCTransferAmount, userBBalAfter.Int64())
+	})
+
+	t.Run("Add Outgoing Ratelimit on ChainA", func(_ *testing.T) {
+		resp, err := query.GRPCQuery[ratelimitingtypes.QueryAllRateLimitsResponse](ctx, chainA, &ratelimitingtypes.QueryAllRateLimitsRequest{})
+		s.Require().NoError(err)
+		s.Require().Nil(resp.RateLimits)
+
+		s.addRateLimit(ctx, chainA, userA, denomA, chanAB.ChannelID, authority.String(), 10, 0, 1)
+
+		resp, err = query.GRPCQuery[ratelimitingtypes.QueryAllRateLimitsResponse](ctx, chainA, &ratelimitingtypes.QueryAllRateLimitsRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(resp.RateLimits, 1)
+
+		// Make transfer again and see the flow has been updated.
+		userABalBefore, err := s.GetChainANativeBalance(ctx, userA)
+		s.Require().NoError(err)
+
+		txResp := s.Transfer(ctx, chainA, userA, chanAB.PortID, chanAB.ChannelID, testvalues.DefaultTransferAmount(denomA), userA.FormattedAddress(), userB.FormattedAddress(), s.GetTimeoutHeight(ctx, chainA), 0, "")
+		s.AssertTxSuccess(txResp)
+
+		packet, err := ibctesting.ParseV1PacketFromEvents(txResp.Events)
+		s.Require().NoError(err)
+		s.Require().NotNil(packet)
+
+		s.Require().Eventually(func() bool {
+			_, err := query.GRPCQuery[chantypes.QueryPacketCommitmentResponse](ctx, chainA, &chantypes.QueryPacketCommitmentRequest{
+				PortId:    chanAB.PortID,
+				ChannelId: chanAB.ChannelID,
+				Sequence:  packet.Sequence,
+			})
+			return err != nil && strings.Contains(err.Error(), "packet commitment hash not found")
+		}, time.Second*70, time.Second)
+
+		userABalAfter, err := s.GetChainANativeBalance(ctx, userA)
+		s.Require().NoError(err)
+
+		// Balanced moved form useA to userB
+		s.Require().Equal(userABalBefore-testvalues.IBCTransferAmount, userABalAfter)
+		userBBalAfter, err := query.Balance(ctx, chainB, userB.FormattedAddress(), ibcTokenB.IBCDenom())
+		s.Require().NoError(err)
+		s.Require().Equal(2*testvalues.IBCTransferAmount, userBBalAfter.Int64())
+
+		// Check the flow has been updated.
+		rateLimit := s.rateLimit(ctx, chainA, denomA, chanAB.ChannelID)
+		s.Require().NotNil(rateLimit)
+		s.Require().Equal(rateLimit.Flow.Outflow.Int64(), testvalues.IBCTransferAmount)
+	})
+
+	t.Run("Fill the quota", func(_ *testing.T) {
+		rateLim := s.rateLimit(ctx, chainA, denomA, chanAB.ChannelID)
+		sendPercentage := rateLim.Quota.MaxPercentSend.Int64()
+
+		// Create an account that can almost exhause the outflow limit.
+		richKidAmt := rateLim.Flow.ChannelValue.MulRaw(sendPercentage).QuoRaw(100).Sub(rateLim.Flow.Outflow)
```

Comment:
Could you not instead create the initial user with enough funds, so that you don't need to create an additional one here?